src/main/java/org/jspecify/annotations/Nullable.java

```diff
@@ -42,88 +42,95 @@
- *       that its uninitialized states can't be observed, it's appropriate to overlook that fact.)
- *   <li>On a <b>type argument</b>: A type usage of "nullable string" appears <i>within</i> the
- *       compound type {@code List<@Nullable String>}. No matter how this type is used (return type,
- *       etc.), this means the same thing: every appearance of {@code E} in {@code List}'s member
- *       signatures will be considered nullable. For a list, this means it may contain null
- *       <i>elements</i>. If the list reference itself might be null as well, we can write
- *       {@code @Nullable List<@Nullable String>}, a "nullable list of nullable strings".
+ *       it might hold a string, and at times it might intentionally hold {@code null}. (Of course,
+ *       any field of a reference type might <i>originally</i> have held {@code null} before it was
+ *       initialized, but as long as the class ensures that its uninitialized states can't be
+ *       observed, it's generally okay to overlook that fact.)
+ *   <li>On a <b>type argument</b>: A type usage of "string-or-null" appears within the compound
+ *       type {@code List<@Nullable String>}. However this type is used (return type, etc.), this
+ *       means the same thing: every appearance of {@code E} in {@code List}'s member signatures
+ *       will be considered nullable (unless it is itself annotated otherwise). For a list, this
+ *       means it may contain null <i>elements</i>. If the list reference itself might be null as
+ *       well, we can write {@code @Nullable List<@Nullable String>}, a "nullable list of nullable
+ *       strings".
  *   <li>On the upper bound of a <b>type parameter</b>: For example, as seen in {@code class List<E
- *       extends @Nullable Object>}. This means that a <i>type argument</i> supplied for that type
- *       parameter is permitted to be nullable if desired: {@code List<@Nullable String>}. (A
- *       non-null type argument, as in {@code List<String>}, is permitted either way.)
- *   <li>On a usage of a <b>type variable</b>: A type parameter, like the {@code E} in {@code
- *       interface List<E>}, defines a "type variable" of the same name, usable only <i>within</i>
- *       the scope of the declaring API element. In any example using {@code String} above, a type
- *       variable like {@code E} might appear instead. {@code @Nullable} continues to mean "or null"
- *       as always, but notably, this works without regard to whether the type argument is
- *       <i>already</i> nullable. For example, suppose that {@code class Foo<E extends @Nullable
- *       Object>} has a method {@code @Nullable E eOrNull()}. Then, whether {@code foo} is of type
+ *       extends @Nullable Object>}. This means that a type argument supplied for that type
+ *       parameter can <i>optionally</i> be nullable: {@code List<@Nullable String>}. (A non-null
```

Comment:
I'm not sure "optionally" does anything to refine "can". Maybe you want to say "can be nullable, but doesn't have to be"?

What about:

> This means that a type argument supplied for that type parameter can be nullable, but doesn't have to be; both `List<@Nullable String>` and `List<String>` are valid.